HEP-event-capture/src/models/Form.ts

```diff
@@ -0,0 +1,64 @@
+import { ProgramStage, DataElement } from "../Dhis2Metadata";
+import { CustomForm } from "../components/CustomForm";
+import * as _ from "lodash";
+
+export interface FormData {
+    sections: FormSection[];
+}
+
+export interface FormSection {
+    title: string;
+    dataElements: DataElement[];
+    programStageId: string;
+}
+
+export const formCodes = {
+    formSection: "HEP_POLICY_SECTION",
+    sectionOrder: "SECTION_ORDER",
+    formDataElementGroup: "HEP_POLICY_EVENT_CAPTURE",
+    optionList: "USE_OPTION_FIELD",
+};
+
+export class Form {
+    public static getFormHtml(programStage: ProgramStage): string {
+        const formData = this.getFormData(programStage);
+        const formHtml = CustomForm({ formData });
+        return formHtml;
+    }
+
+    public static getFormData(programStage: ProgramStage): FormData {
+        const allDataElementGroups = programStage.programStageDataElements.map(
+            psde =>
+                psde.dataElement.dataElementGroups.filter(deg =>
+                    new RegExp("^" + formCodes.formSection).test(deg.code)
+                )[0]
+        );
+        const dataElementGroups = _.uniqBy(allDataElementGroups, "code");
+
+        const sections = dataElementGroups.map(deg => {
+            const orderAttribute = deg.attributeValues.find(
+                av => av.attribute.code === formCodes.sectionOrder
+            );
+            const order = orderAttribute && orderAttribute.value;
+            const formattedDataElements = deg.dataElements.map(de => ({
+                ...de,
+                useOptionList: de.attributeValues.some(
```

Comment:
From your comment: "I added an attribute to the dataElements to specify if their inputs should be OptionList (comboxes) or regular TextFields". If the value cannot be calculated from their OptionsList, it makes sense, I leave this note hust to validate with @ifoche, since adding attributes should be the last resort.